packages/@uppy/companion/src/server/provider/webdav/index.js

```diff
@@ -72,6 +73,39 @@ export default class WebdavProvider extends Provider {
     }
   }
 
+  async resolveClientFromPublicURL({ baseURL, publicLinkToken }) {
+    const basewebdavURL = baseURL.replace('/index.php', '')
+    const authOptions = {
+      authType: AuthType.Password,
+      username: publicLinkToken,
+      password: 'null',
+    }
+    // possible webdav configs to try
+    const webDavConfigs = [
+      { url: `${basewebdavURL}/public.php/webdav/`, ...authOptions },
+      {
+        url: `${basewebdavURL}/remote.php/dav/public-files/${publicLinkToken}`,
+        ...authOptions,
+      },
+      {
+        url: `${basewebdavURL}/dav/public-files/${publicLinkToken}`,
+        ...authOptions,
+      },
+    ]
+
+    let error
+    for (const config of webDavConfigs) {
+      try {
+        const client = await this.getClientHelper(config)
+        await client.exists(defaultDirectory)
+        return { client, config }
+      } catch (err) {
+        error = err
```

Comment:
this doesn't any sense, you can just omit the whole `try`/`catch` if you do nothing with the error